### PR TITLE
[BUILD] Skip copying spark extensions artifacts during make distribution

### PR DESCRIPTION
**Pull request by pan3793**
# :mag: Description
## Issue References 🔗

Currently, the `build/dist` always tries to copy all versions of Kyuubi Spark extensions from those modules' `target` folder, which results in an undetermined output.

Let's say we perform `build/dist -Pspark-3.1`, change some code, then perform `build/dist -Pspark-3.3`, which would result in both Spark 3.1 and 3.3 extensions being included in the final binary tarball, specifically, the 3.1's artifact is outdated.

This also violates the principle of "repeatable building steps".

## Describe Your Solution 🔧

Simplify skipping copying those jars, and users can download them from Maven Central or build by themselves like other Kyuubi Spark plugins do.

## Types of changes :bookmark:

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

#### Behavior Without This Pull Request :coffin:

`build/dist` may produce undetermined outputs (may include outdated Spark extension jars)

#### Behavior With This Pull Request :tada:

`build/dist` always produces determined outputs (no Spark extension jars)

---

# Checklists
## 📝 Author Self Checklist

- [x] My code follows the [style guidelines](https://kyuubi.readthedocs.io/en/master/contributing/code/style.html) of this project
- [x] I have performed a self-review
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] This patch was not authored or co-authored using [Generative Tooling](https://www.apache.org/legal/generative-tooling.html)

## 📝 Committer Pre-Merge Checklist

- [ ] Pull request title is okay.
- [ ] No license issues.
- [ ] Milestone correctly set?
- [ ] Test coverage is ok
- [ ] Assignees are selected.
- [ ] Minimum number of approvals
- [ ] No changes are requested

**Be nice. Be informative.**

### CODE REVIEW

**Comment by wForget** [comment]
I prefer to package it into distribution so we do not need to do extra work, we can clean extensions directory first in dist script.

File: build/dist [hunk 1]
```diff
@@ -338,16 +338,6 @@ if [[ -f "$KYUUBI_HOME/tools/spark-block-cleaner/target/spark-block-cleaner_${SC
   cp "$KYUUBI_HOME/tools/spark-block-cleaner/target/spark-block-cleaner_${SCALA_VERSION}-${VERSION}.jar" "$DISTDIR/tools/spark-block-cleaner/jars/"
 fi
 
-# Copy Kyuubi Spark extension
-SPARK_EXTENSION_VERSIONS=('3-1' '3-2' '3-3' '3-4' '3-5')
```
[the remark this answers is not in the document]
Comment (ulysses-you) [reply build/dist:342]: yeah, I agree. Make the target version distribution sounds good to me.

[the remark this answers is not in the document]
Comment (wForget) [reply build/dist:342]: I think this is already the current behavior. Only one version is compiled when -Pspark-XXX is specified.

[the remark this answers is not in the document]
Comment (wForget) [reply build/dist:342]: > I think this is already the current behavior. Only one version is compiled when -Pspark-XXX is specified.

Sorry, my understanding is wrong.

[the remark this answers is not in the document]
Comment (pan3793) [reply build/dist:342]: a workaround is performing `rm -rf **/target` at the beginning at `build/dist`.

then if users build with `-Pspark-3.3`, only 3.3 extension will be shipped, but nothing if without spark profiles.

How do you think this way?

[the remark this answers is not in the document]
Comment (wForget) [reply build/dist:342]: We can get the spark binary version like this and then only copy the specified version:

```
SPARK_BINARY_VERSION=$("$MVN" help:evaluate -Dexpression=spark.binary.version $@ 2>/dev/null\
    | grep -v "INFO"\
    | grep -v "WARNING"\
    | tail -n 1)
```

[the remark this answers is not in the document]
Comment (pan3793) [reply build/dist:342]: we didn't compile the extension when no `-Pspark-x.y` is provided

[the remark this answers is not in the document]
Comment (wForget) [reply build/dist:342]: so check whether it exists before copying?

**Pull request closed without merging**